src/transformers/data/data_collator.py

```diff
@@ -435,6 +443,11 @@ def _whole_word_mask(self, input_tokens: List[str], max_predictions=512):
         """
         Get 0/1 labels for masked tokens with whole word mask proxy
         """
+        if not isinstance(self.tokenizer, (BertTokenizer, BertTokenizerFast)):
+            warnings.warn(
+                "DataCollatorForWholeWordMask is only suitable for BertTokenizer-like tokenizers."
+                "Please refer to the documentation for more information."
+            )
```

Comment:
Maybe just raise an error since it's not supposed to be used in that case?

Reply:
I thought about this, but settled for a warning, because I wasn't sure whether there are other tokenizers that do not inherit from `BertTokenizer`/`BertTokenizerFast` but are nevertheless compatible.